src/goto-instrument/contracts/contracts.cpp

```diff
@@ -771,36 +785,34 @@ void code_contractst::apply_function_contract(
   is_fresh.create_declarations();
   is_fresh.add_memory_map_decl(new_program);
 
-  // Insert assertion of the precondition immediately before the call site.
-  exprt::operandst requires_conjuncts;
-  for(const auto &r : type.requires())
+  // Generate: assert(requires)
+  for(const auto &clause : type.requires())
   {
-    requires_conjuncts.push_back(
-      to_lambda_expr(r).application(instantiation_values));
-  }
-  auto requires = conjunction(requires_conjuncts);
-  requires.add_source_location() =
-    requires_conjuncts.empty() ? type.source_location()
-                               : type.requires().front().source_location();
-  if(!requires.is_true())
-  {
-    if(has_subexpr(requires, ID_exists) || has_subexpr(requires, ID_forall))
-      add_quantified_variable(requires, mode);
-
-    goto_programt assertion;
-    converter.goto_convert(code_assertt(requires), assertion, mode);
-    assertion.instructions.back().source_location_nonconst() =
-      requires.source_location();
-    assertion.instructions.back().source_location_nonconst().set_comment(
-      "Check requires clause");
-    assertion.instructions.back().source_location_nonconst().set_property_class(
-      ID_precondition);
-    is_fresh.update_requires(assertion);
-    new_program.destructive_append(assertion);
+    auto instantiated_clause =
+      to_lambda_expr(clause).application(instantiation_values);
+
+    if(!instantiated_clause.is_true())
+    {
+      source_locationt _location = clause.source_location();
+      _location.set_line(location.get_line());
+      _location.set_comment(std::string("Check requires clause of ")
```

Comment:
could we also embed the string representation of the formula being asserted in the source location ?

Reply:
See https://github.com/diffblue/cbmc/pull/7159#discussion_r984978914